packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -705,59 +705,78 @@ export class BackgroundScript {
 
   private async sendTip(): Promise<{ success: boolean }> {
     const tabId = this.activeTab
-    const streamId = this.assoc.getStreamId({ tabId, frameId: 0 })
-    if (!streamId) {
-      this.log('can not find top frame for tabId=%d', tabId)
-      return { success: false }
-    }
-    const stream = this.streams.getStream(streamId)
-    const token = this.auth.getStoredToken()
 
-    // TODO: return detailed errors
-    if (!stream || !token) {
-      this.log(
-        'sendTip: no stream | token. !!stream !!token ',
-        !!stream,
-        !!token
-      )
-      return { success: false }
-    }
+    const streams = Object.entries(this.assoc.getTabStreams(tabId))
+    const numTips = streams.length
+    const success: boolean[] = []
+    const messages: TipSent[] = []
+
+    for (const [frameIdStr, streamId] of streams) {
```

Comment:
If one was to do this properly, one should loop through the streams and perhaps wait until the exchange rate was set ( in the case of streams that are just activating - note that with iframes - they tend to connect in a staggering manner) 

Another thing to consider is that in the UI currently the tip button is only shown  if the stream has activated (so the exchange rate is shown). Would need to consider multiple frames.